foundry.md

```diff
@@ -129,6 +129,39 @@ This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)`
          </account>
 ```
 
+
+#### `etch` - Sets the code of an account.
+
+```
+function etch(address who, bytes calldata code) external;
+```
+
+`call.etch` will match when the `etch` function is called at the [Foundry cheatcode address](https://book.getfoundry.sh/cheatcodes/#cheatcodes-reference).
+This rule then takes the account using `#asWord(#range(LM, ARGSTART +Int 4, 32)` and the new bytecode using `#range(LM, ARGSTART +Int 36, 32)` and forwards them to the `#setCode` marker which updates the account accordingly.
+
+```k
+    rule [call.etch]:
+         <k> CALL _ CHEAT_ADDR 0 ARGSTART _ARGWIDTH 0 0
+          => #setCode(#asWord(#range(LM, ARGSTART +Int 4, 32)), #range(LM, ARGSTART +Int 36, 32))
```

Comment:
The `code` should be read from the `calldata` cell instead of the `localmem`. The length can be `>32` bytes